### PR TITLE
Disable Arrow serialization for Spark JDBC driver

### PR DESCRIPTION
**Pull request by wendigo**

### CODE REVIEW

**Review by ebyhr — APPROVED** [review]

**Comment by ebyhr** [comment]
I'm not sure why #13183 didn't trigger Databricks product test. Running the test https://github.com/trinodb/trino/runs/7350924334 forcibly in #13185

File: testing/trino-product-tests-launcher/src/main/resources/docker/presto-product-tests/conf/environment/singlenode-delta-lake-databricks/tempto-configuration.yaml [hunk 1]
```diff
@@ -7,7 +7,7 @@ databases:
     prepare_statement:
       - USE ${databases.delta.schema}
     table_manager_type: jdbc
-    jdbc_url: ${DATABRICKS_JDBC_URL}
+    jdbc_url: ${DATABRICKS_JDBC_URL};EnableArrow=0
```
Comment (findinpath) [inline testing/trino-product-tests-launcher/src/main/resources/docker/presto-product-tests/conf/environment/singlenode-delta-lake-databricks/tempto-configuration.yaml:10]: Please update the `DATABRICKS_JDBC_URL` secret instead.
We have now proof that it works with `Arrow` disabled, so we can safely update the secret.

Comment (ebyhr) [reply testing/trino-product-tests-launcher/src/main/resources/docker/presto-product-tests/conf/environment/singlenode-delta-lake-databricks/tempto-configuration.yaml:10]: I disagree with putting the option in secret. People who can see `DATABRICKS_JDBC_URL` value is limited.

Comment (findinpath) [reply testing/trino-product-tests-launcher/src/main/resources/docker/presto-product-tests/conf/environment/singlenode-delta-lake-databricks/tempto-configuration.yaml:10]: This is true. Should I make eventually a new PR which exposes the full jdbc URL and hides only the Databricks components:
- organisation
- account
- cluster

?

**Comment by ebyhr** [comment]
Merged, thanks!

**Pull request merged**